packages/commons/src/auth/authData.ts

```diff
@@ -1,54 +1,163 @@
-import { JwtPayload } from "jsonwebtoken";
-import { TenantId } from "pagopa-interop-models";
+import { TenantId, unsafeBrandId } from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
 import { z } from "zod";
 
-export const userRoles = {
+const uiTokenUserRoles = {
   ADMIN_ROLE: "admin",
   SECURITY_ROLE: "security",
   API_ROLE: "api",
+  SUPPORT_ROLE: "support",
+} as const;
+
+const UIUserRole = z.enum([
+  Object.values(uiTokenUserRoles)[0],
+  ...Object.values(uiTokenUserRoles).slice(1),
+]);
+
+export const userRoles = {
+  ...uiTokenUserRoles,
   M2M_ROLE: "m2m",
   INTERNAL_ROLE: "internal",
-  SUPPORT_ROLE: "support",
 } as const;
 
-export const UserRoles = z.enum([
+export const UserRole = z.enum([
   Object.values(userRoles)[0],
   ...Object.values(userRoles).slice(1),
 ]);
-export type UserRoles = z.infer<typeof UserRoles>;
+export type UserRole = z.infer<typeof UserRole>;
 
-export const AuthJWTToken = z.object({
-  organizationId: TenantId,
-  "user-roles": z
-    .string()
-    .optional()
-    .transform((val) => val?.split(",")),
-  role: z
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const CommaSeparatedStringToArray = <T extends z.ZodType>(t: T) =>
+  z
     .string()
-    .optional()
-    .transform((val) => val?.split(",")),
-  uid: z.string().uuid().optional(),
-  organization: z.object({
-    roles: z.array(
-      z.object({
-        role: z.string(),
-      })
-    ),
-  }),
-  externalId: z.object({
-    origin: z.string(),
-    value: z.string(),
-  }),
+    .nonempty()
+    .transform((s: string) => s.split(","))
+    .pipe(z.array(t));
+
+const SharedJWTClaims = z.object({
+  // All standard claims except "sub", which is not present in UI tokens
+  iss: z.string(),
+  aud: CommaSeparatedStringToArray(z.string()),
+  exp: z.number(),
+  nbf: z.number(),
+  iat: z.number(),
+  jti: z.string().uuid(),
 });
-export type AuthJWTToken = z.infer<typeof AuthJWTToken> & JwtPayload;
 
+export const M2MAuthToken = SharedJWTClaims.merge(
+  z.object({
+    role: z.literal("m2m"),
+    organizationId: z.string().uuid(),
+    client_id: z.string().uuid(),
```

Comment:
@beetlecrunch is the `client_id` claim in m2m token?

Reply:
Yes, it's correct.